job-runner/chain/chain.go

```diff
@@ -96,108 +117,167 @@ func (c *chain) NextJobs(jobId string) proto.Jobs {
 	return nextJobs
 }
 
-// JobIsReady returns whether or not a job is ready to run. A job is considered
-// ready to run if all of its previous jobs are complete. If any previous jobs
-// are not complete, the job is not ready to run.
-func (c *chain) JobIsReady(jobId string) bool {
-	c.jcMux.RLock()
-	defer c.jcMux.RUnlock()
-	for _, job := range c.previousJobs(jobId) {
-		if job.State != proto.STATE_COMPLETE {
-			return false
+// IsRunnable returns whether or not a job is runnable. A job is considered
+// runnable if it is Pending or Stopped with some retry attempts remaining,
+// and all of its previous jobs are complete. If any previous jobs are not
+// complete, the job is not runnable.
+func (c *Chain) IsRunnable(jobId string) bool {
+	c.jobsMux.RLock()
+	defer c.jobsMux.RUnlock()
+	return c.isRunnable(jobId)
+}
+
+// RunnableJobs returns a list of all jobs that are runnable. A job is
+// runnable if all of its previous jobs are complete and it is Pending
+// or Stopped with some retries still remaining.
+func (c *Chain) RunnableJobs() proto.Jobs {
+	// Loop through every job in the chain and check if it's runnable.
+	var runnableJobs proto.Jobs
+	for jobId, job := range c.jobChain.Jobs {
+		if !c.IsRunnable(jobId) {
+			continue
 		}
+
+		runnableJobs = append(runnableJobs, job)
 	}
-	return true
+
+	return runnableJobs
 }
 
-// IsDone returns two booleans - the first one indicates whether or not the
-// chain is done, and the second one indicates whether or not the chain is
-// complete.
+// IsDoneRunning returns two booleans - the first indicates whether the chain
+// is done, and the second indicates whether the chain is complete.
 //
-// A chain is done running if there are no more jobs in it that can run. This
-// can happen if all of the jobs in the chain or complete, or if some or all
-// of the jobs in the chain failed.
+// A chain is done running if there are no more jobs in it that can be run or
```

Comment:
"A chain is done running if there are no jobs in it running and there are no more jobs in it that can be run." is more clear to me